app/services/books/index.rb

```diff
@@ -52,8 +63,27 @@ def exists?
       OsElasticsearchClient.instance.indices.exists?(index: name)
     end
 
+    def not_exists?
+      !OsElasticsearchClient.instance.indices.exists?(index: name)
+    end
+
     private
 
+    def wait_until(this_happens)
+      tries = 1
+      until self.send(this_happens) || tries > WAIT_UNTIL_MAX_TRIES  do
+        Rails.logger.debug("Books::Index. Waiting #{WAIT_UNTIL_TRIES_INTERVAL_SEC} secs for #{name} to #{this_happens.to_s}, num_tries so far: #{tries}")
+        sleep(WAIT_UNTIL_TRIES_INTERVAL_SEC)
+        tries += 1
+      end
+
+      if tries >= WAIT_UNTIL_MAX_TRIES
+        raise IndexResourceNotReadyError.new("Books::Index. #{name}:#{this_happens.to_s} failed after #{tries} tries.")
+      end
```

Comment:
This could be at the end of the `until` loop above (changing `>=` to `>` probably which would let you remove the `|| tries > WAIT_UNTIL_MAX_TRIES` there.